pkg/storegateway/batch_series_test.go

```diff
@@ -650,6 +651,8 @@ func TestSeriesSetWithoutChunks(t *testing.T) {
 func TestPreloadingBatchSet(t *testing.T) {
 	test.VerifyNoLeak(t)
 
+	const delay = 10 * time.Millisecond
```

Comment:
Note to reviewers: I've added a delay when iterating the underlying sets so that it's more realistic.